### DISCUSSION TITLE
Source Generators execution order

### DISCUSSION THREAD

**Discussion by lentrodev** [dotnet/roslyn#57912] [Show and tell]
If anyone need to know how to reorder source generators, located in different assemblies, or if you need to be sure your generators will be executed after others, just reorder `Project/PackageReference`s in your target `csproj` file.
For example, we have project `Test`, and two projects with source generators, `Test.A` and `Test.B`, so our `Test.csproj` file will look similar to `this:`

```xml
<ItemGroup>
    <ProjectReference Include="..\Test.A\Test.A.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false"/>
    <ProjectReference Include="..\Test.B\Test.B.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false"/>
</ItemGroup>
```

Here, firstly will be executed generators from the `Test.A` project, and then from the `Test.B`.

Also, will be nice to have feature to order generators in the assemblies, but there's no way to do that at the moment.

**Comment by jasonmalinowski** [comment]
I'm a bit confused by this post -- even though that may control the order in the sense of "which one do we run first", generators will see the same input so the ordering isn't observable short of a debugger....

**Reply by lentrodev** [reply]
Well, then nevermind what I wrote, seems like I didn't notice that and ran to make this post, xD.

**Reply by RikkiGibson** [reply]
That said, it seems like generator ordering is a common question and something that we are thinking quite a bit about how to solve.

One of the big issues is that whenever someone comes to us with this ask, they typically say "I need to run before all the other generators" or "I need to run after all the generators" and it's problematic when more than one generator needs this 😉
